langchain4j-redis/src/main/java/dev/langchain4j/store/embedding/redis/RedisEmbeddingStore.java

```diff
@@ -169,9 +173,19 @@ private void addAllInternal(List<String> ids, List<Embedding> embeddings, List<T
                 Map<String, Object> fields = new HashMap<>();
                 fields.put(schema.vectorFieldName(), embedding.vector());
                 if (textSegment != null) {
-                    // do not check metadata key is included in RedisSchema#metadataKeys
+                    // NOTE: do not check metadata key is included in RedisSchema#metadataKeys
                     fields.put(schema.scalarFieldName(), textSegment.text());
-                    fields.putAll(textSegment.metadata().asMap());
+                    Map<String, Object> metadataMap = textSegment.metadata().toMap();
+
+                    // NOTE: since Java types and Redis types don't always correspond, there is a conversion to be done here.
+                    for (Map.Entry<String, SchemaField> fieldEntry : schema.schemaFieldMap().entrySet()) {
+                        String metadataKey = fieldEntry.getKey();
+                        SchemaField field = fieldEntry.getValue();
+                        if (field instanceof TextField && metadataMap.containsKey(metadataKey)) {
```

Comment:
Could you please provide an example when this conversion is necessary?

Reply:
Is it because of this?

> It seems that the Redis NumericType range is different from the Java range, so I'm treating Long and Double as TextField.

Reply:
> Is it because of this?
> 
> > It seems that the Redis NumericType range is different from the Java range, so I'm treating Long and Double as TextField.

Yes. When I treat `Long` and `Double` as `NumericType` in Redis, it seems that `Redis` can't handle it in a right way, and the exception will occur:

```java
java.lang.NumberFormatException: For input string: "9.22337203685e+18"

	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
	at java.base/java.lang.Long.parseLong(Long.java:708)
	at java.base/java.lang.Long.parseLong(Long.java:831)
	at dev.langchain4j.data.document.Metadata.getLong(Metadata.java:201)
```